transform/transform.go

```diff
@@ -116,13 +116,19 @@ func (ti *transformInstruction) transform(in interface{}, fieldType string, modi
 // trransformInstructions defines a set of instructions and a method for combining their results.
 // The default method is to take the first non-nil result.
 type transformInstructions struct {
-	From   []*transformInstruction `json:"from"`
-	Method transformMethod         `json:"method"`
+	From          []*transformInstruction `json:"from"`
+	Method        transformMethod         `json:"method"`
+	MethodOptions methodOptions           `json:"methodOptions"`
 }
 
 type transformInstructionsJSON struct {
-	From   []*transformInstruction `json:"from"`
-	Method string                  `json:"method"`
+	From          []*transformInstruction `json:"from"`
+	Method        string                  `json:"method"`
+	MethodOptions methodOptions           `json:"methodOptions"`
+}
+
+type methodOptions struct {
+	ConcatenateDelimiter interface{} `json:"concatenateDelimiter"`
```

Comment:
That should be a string not an empty interface, it makes the code nicer. Also though concat can theoretically work on slices or other types I think it is fine to say that the delimiter is only supported for strings.